openghg/transform/flux/_edgar.py

```diff
@@ -63,210 +204,96 @@ def parse_edgar(
 
     if len(date) == 4:
         year = int(date)
     else:
-        raise ValueError(f"Do no accept date which does not represent a year yet: {date}")
+        raise ValueError(
+            f"Date {date} does not represent a year;" " only annual EDGAR data can be processed currently."
+        )
 
-    files_by_year = {}
-    for file in data_files:
+    FileInfo = namedtuple("FileInfo", "path metadata")
+    files_by_year: dict[int, FileInfo] = {}
+    for data_file in data_files:
         try:
-            file_info = _extract_file_info(file)
+            metadata = assemble_edgar_metadata(data_file, species, edgar_version)
         except ValueError:
             continue
+        else:
+            # Check if data is actually monthly "...2015_1" etc. - can't parse yet
+            if "month" in metadata:
+                raise NotImplementedError("Unable to parse monthly EDGAR data at present.")
 
-        # Check if data is actually monthly "...2015_1" etc. - can't parse yet
-        if "month" in file_info:
-            raise NotImplementedError("Unable to parse monthly EDGAR data at present.")
-
-        year_from_file = file_info["year"]
+            files_by_year[metadata["year"]] = FileInfo(data_file, metadata)
 
-        files_by_year[year_from_file] = file
+    if not files_by_year:
+        raise ValueError(f"Unable to extract EDGAR file info from any files in {datapath}.")
 
-        if year_from_file == year:
-            edgar_file = file
-            edgar_file_info = file_info
-            break
-    else:
-        all_years = list(files_by_year.keys())
-        all_years.sort()
+    try:
+        edgar_file, edgar_file_info = files_by_year[year]
+    except KeyError:
+        all_years = sorted(list(files_by_year.keys()))
         start_year, end_year = all_years[0], all_years[-1]
+
         if year < start_year:
-            raise ValueError(
-                f"EDGAR {edgar_version} range: {start_year}-{end_year}." f" {year} is before this period."
-            )
+            raise ValueError(f"Files span range: {start_year}-{end_year}." f" {year} is before this period.")
         elif year > end_year:
-            logger.info(
-                f"Using last available year from EDGAR {edgar_version} range:" f"{start_year}-{end_year}."
+            logger.info(f"Using last available year from range:" f"{start_year}-{end_year}.")
+        edgar_file, edgar_file_info = files_by_year[end_year]
+
+    species_label = edgar_file_info["species"]
+    version = edgar_file_info["version"]
+
+    # get dataset
+    # using .open("rb") for pathlib.Path and zipfile.Path compatibility
+    edgar_ds = xr.open_dataset(edgar_file.open("rb"))
+
+    # Expected name e.g. "emi_ch4", "emi_co2" for version <= 7; "fluxes" for version 8
+    name = "fluxes" if version.startswith("v8") else f"emi_{species_label}"
+
+    # just in case we didn't get the name right...
+    if name not in (dvs := edgar_ds.data_vars):
```

Comment:
Related to Rachel's suggestion above, I think this can just be
```suggestion
    if name not in edgar_ds.data_vars:
        if version.startswith("v8"):
            raise ValueError(...)
```